### PR TITLE
add and remove functionality added

### PR DESCRIPTION
**Pull request by ZabihullahNooriWardak**
In this branch I did the following works : 
 1 : I added the Add functionality .
 2 : I added the remove functionality .
 3 : Edit button Function Added

### CODE REVIEW

File: src/modules/functions.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+const txtField = document.getElementById('taskTextField');
+const taskContainerFromhtml = document.querySelector('.tasks');
+let removeButton = [];
+
+function add(arr) {
+  if (localStorage.getItem('listTasks') !== null) {
+    arr = JSON.parse(localStorage.getItem('listTasks'));
+  }
+  arr.push(txtField.value);
+  localStorage.setItem('listTasks', JSON.stringify(arr));
```
Comment (tugsem) [inline src/modules/functions.js]: - [ ] Please store the objects as per the requirement of this milestone, you only stored the text value, but we should also show the index property starting from 1, 2, 3... and the completed property which should have boolean values. In the next capstone, you are also going to implement the checkbox functionality which will update the objects in the local storage.

File: src/modules/functions.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
+/* eslint-disable no-restricted-globals */
+// add button function
+let listTasks = [];
+if (localStorage.getItem('listTasks') !== null) {
+  listTasks = JSON.parse(localStorage.getItem('listTasks'));
+}
+// DOM manipulation
+const addbtn = document.querySelector('.add');
+const msg = document.querySelector('.msg');
+const txtField = document.getElementById('taskTextField');
```
Comment (tugsem) [inline src/modules/functions.js:13]: - [ ] As per the requirement below, please add a method for editing the task decriptions.
image

**Review by tugsem — CHANGES_REQUESTED** [review]
##  STATUS: REQUIRED CHANGES ♻️
Hello @ZabihullahNooriWardak 👋 , 

Good job so far! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text
## Highlights 🚀 
✔️local storage implementation is correct
✔️hardcoded data is removed
✔️add and delete methods are added
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/functions.js (hunk 2 above)
Comment (brytebee) [reply src/modules/functions.js:13]: 👏👏👏👏

You have correctly implemented the edit function.

File: src/modules/functions.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  arr.push({
+    isCompleted: false,
+    index: arr.length,
+    description: txtField.value,
+  });
+  localStorage.setItem('listTasks', JSON.stringify(arr));
+}
+
+// For displaying all the Task in the Screen.
+function display(arr) {
```
Comment (brytebee) [inline src/modules/functions.js:36]: Nice work on coming this far.

There are still some changes to be made.

- On the initial render of the app, the indexes begin at zero, it should begin at 1. So in your add function, let index be `arr.length + 1`.
- When one is deleted, an index is duplicated, the fix above should also fix this.

image

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @ZabihullahNooriWardak 👋, 

##

### Highlights💡
- You have correctly implemented the edit function.✔️
- Great work saving all changes to local storage.✔️
- Good commit message.✔️ 
- Good PR Summary.✔️

##

Great work on making the changes 👏 👏 
You've done well implementing some of the requested changes requested by the previous review, but there are still some which aren't addressed yet.


# Suggested changes
_Check the comments under the review._




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by prolajumokeoni — APPROVED** [review]
Hi @ZabihullahNooriWardak 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
  - App index updated ✅ 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**